README.md

```diff
@@ -18,9 +22,64 @@ If you'd prefer to use a local copy of the portage tree, you can point
+breakage, but they are by no means exhaustive. Be especially careful with packages that
+might affect `sys-devel/binutils`, `sys-devel/gcc`, `sys-kernel/linux-headers`, and
+`sys-libs/glibc` (see `TOOLCHAIN_PKGS` in `scripts/build_library/toolchain_util.sh`).
+
+## git log
+
+`git log <category>/<package>` will show commits that touched that directory. These can give
+clues about why a package was included in the first place and where to look to ensure it really
+is unused.
+
+## equery d
+
+`equery d -a <package-name>` will tell you what packages depend on `package-name`. It will
+also generate a lot of false positives, since it considers all dependencies for all use flags,
+even ones we do not use, such as `test`.
+
+## emerge --emptytree
+
+`emerge --pretend --verbose --emptytree <package-name>` _should_ give a list of all the dependencies
+for a given package. After removing the ebuild and package, make sure its removal doesn't break
+the packages `board-packages`, `sdk-depends`, and `@system`. Remember to use the `emerge-<arch>-usr`
+commands to check `board-packages` and `emerge` to check `sdk-depends`. Use both when checking
+`@system`.
+
+The ebuild must be removed from `portage-stable` _and_ the package must be removed locally. If only the
+ebuild is removed, the package will be silently elided in the dependency list. To see if there are any
+packages installed without ebuilds run `eix -tTc`. There are no `eix-<arch>-usr` wrappers, so double
+check the packages are also unmerged via the `emerge-<arch>-usr` commands. Make sure to run
+`eix-update` before running other `eix` commands.
+
+When making changes affecting virtual packages, `--emptytree` will always prefer a already installed
+provider (provided it is not masked or otherwise disabled), so unmerge the current provider before
+running `emerge --emptytree` to ensure the virtuals are resolved correctl. See
+[this bug](https://bugs.gentoo.org/127956).
+
+## grep, git grep, repo grep, ripgrep, find, etc
+
+Use your favorite grep variant to see if the package is used anywhere. Good places to double check are
+coreos-overlay, manifest, scripts, and portage-stable, as well as anything specific to the package.
+
+Be sure to check `coreos-overlay` to ensure there are no use flags, accept_keywords, or other bits
+relating to the package being removed left over.
+
+## Updating the metadata cache
+
 If you remove a package, make sure to delete the corresponding files in
 metadata/md5cache, or run use egencache to do it for you:
+```
+    egencache --update --repo portage-stable
+```
+There is also the `update_metadata` script which will update both `portage-stable` and `coreos-overlay`
+and optionally generate a commit message.
 
-    egencache --update portage-stable
-
-Upstream Gentoo sources: http://sources.gentoo.org/gentoo-x86/
+## Testing changes
+If you work at CoreOS, or have Jenkins running with our configuration, you can make pull requests with your changes and
```

Comment:
The first part isn't necessary. `If you're running the [jenkins-os](https://github.com/coreos/jenkins-os) Jenkins configuration, ...`